### ISSUE TITLE
Oozie does not initialize on HA

### ISSUE THREAD

**Issue by adam-stasiak** [GoogleCloudPlatform/dataproc-initialization-actions#211]
I tried to create HA cluster using default oozie.sh script from the repo and failed because of:
```+ cp -ax share /usr/lib/oozie/
+ hadoop fs -mkdir -p /user/oozie/
18/03/06 10:57:18 INFO gcs.GoogleHadoopFileSystemBase: GHFS version: 1.6.3-hadoop2
+ hadoop fs -put share/ /user/oozie/
18/03/06 10:57:21 INFO gcs.GoogleHadoopFileSystemBase: GHFS version: 1.6.3-hadoop2
put: `/user/oozie/share/lib/distcp/hadoop-distcp-2.8.2.jar': File exists
put: `/user/oozie/share/lib/distcp/oozie-sharelib-distcp-4.3.0.jar': File exists
put: `/user/oozie/share/lib/hcatalog/ant-1.9.1.jar': File exists
```

One of my tries to solve this issue was skipping PUT step if there is something inside /user/oozie: 
```++ hdfs dfs -count /user/oozie
++ awk '{print $2}'
18/03/06 09:57:36 INFO gcs.GoogleHadoopFileSystemBase: GHFS version: 1.6.3-hadoop2
+ [[ 385 -eq 0 ]]
+ rm -rf ext-2.2 ext-2.2.zip core-site-patch.xml share oozie-sharelib.tar.gz
+ systemctl restart hadoop-hdfs-namenode
+ systemctl restart hadoop-hdfs-secondarynamenode
Failed to restart hadoop-hdfs-secondarynamenode.service: Unit hadoop-hdfs-secondarynamenode.service failed to load: No such file or directory.
```

For Single and Standard config script works fine - On 1.2.0 Image Version operation succeded.

**Comment by adam-stasiak** [comment]
The same issue on 1.2.27.

**Comment by karth295** [comment]
On a non-HA cluster, the master runs a namenode and secondary namenode. On an HA cluster, you have two regular namenodes (on different hosts) that coordinate through zookeeper. On dataproc, we put namenodes on -m-0 and -m-1. The third master (-m-2) seems to have the NN package installed, but it is not enabled in systemd. HDFS does not allow us to run 3 namenodes until Hadoop 3. All masters in an HA cluster have a resource manager, so that one is not a problem.

Background: https://hadoop.apache.org/docs/current/hadoop-project-dist/hadoop-hdfs/HDFSHighAvailabilityWithQJM.html, https://hadoop.apache.org/docs/current/hadoop-yarn/hadoop-yarn-site/ResourceManagerHA.html.

I would suggest doing something like `for each service: if service is enabled in systemd: restart it`

Side note: configuring HA Oozie is a little complicated because it requires an HA database: https://www.cloudera.com/documentation/enterprise/5-6-x/topics/cdh_hag_oozie_ha.html.

**Comment by adam-stasiak** [comment]
Issue solved with:
fix for intialization on HA cluster #271
